### PR TITLE
2432+2427/ib model creation and dashboard

### PR DESCRIPTION
**Pull request by Caleb-T-Owens**
## Status

* Current Status: WIP; ~Ready for (front-end / tech) review~
* Review App link(s): 
1. https://teachcomputing-pr-1751.herokuapp.com/dashboard
2. https://teachcomputing-pr-1751.herokuapp.com/certificate/i-belong-certificate
* Related to 
  - https://github.com/NCCE/teachcomputing.org-issues/issues/2432
  - https://github.com/NCCE/teachcomputing.org-issues/issues/2427

## Review progress:

- [x] Browser tested
- [ ] Front-end review completed
- [x] Tech review completed

## What's changed?

- new panels on main dashboard
- new certificate/i-belong-certificate certificate dashboard with progress tracking
- new activities
- new programme activity groupings, with completion requirements
- ability to complete an achievement without submitting evidence (only applies to the 3 new "Access the following resources to support you" activities)

## Steps to perform after deploying to production

1. Add I Belong credly badge to DB: https://github.com/NCCE/teachcomputing.org-issues/issues/2431
2. When business is ready, and happy with remaining parts of I Belong programme, turn on the feature by setting the environment variable 
```shell
FLAG_IBELONG_PROGRAMME="on"
```

### CODE REVIEW

File: app/components/community_activity_component/community_activity_component.html.erb [hunk 1]
```diff
@@ -14,31 +14,38 @@
...
+                    <%= text_field_tag "#{:self_verification_info.to_s}_#{@activity.id}", '', class: 'govuk-input', name: :self_verification_info %>
+                  <% end %>
+                  <%= f.hidden_field :activity_id, value: @activity.id %>
+                </span>
+                <%= f.submit t('.evidence_submit_label'), class: 'govuk-button button button--focus-white' %>
+              <% end %>
+            </span>
           </span>
-        </span>
+        <% end %>
```
Comment (dcorking) [inline app/components/community_activity_component/community_activity_component.html.erb:48]: The logic here might be easier for a maintainer to follow if you extract the branch bodies as 2 partials: _verify_achievement_without_evidence _verify_achievement_with_evidence .

File: app/webpacker/stylesheets/components/_card.scss [hunk 2]
```diff
@@ -257,14 +257,18 @@
   justify-content: space-between;
 }
 
-.card--dashboard--cs-accelerator,
+.card--dashboard--accelerator,
```
Comment (dcorking) [inline app/webpacker/stylesheets/components/_card.scss:260]: was this change deliberate? I think this class is used in /app/views/dashboard/certificates/_cs-accelerator.html.erb

File: db/seeds/programme_activity_groupings/i_belong_certificate.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
+i_belong = Programme.i_belong_certificate
+
+puts 'Creating Programme Activity Groupings'
+
+## The numbering of the groupings starts at 2 for historical reasons: group_one with sort_key 1 existed when users were required
+## to complete 2 courses, one from each of groups 1 and 2.
```
Comment (dcorking) [inline db/seeds/programme_activity_groupings/i_belong_certificate.rb:6]: (I'm sad that this funky numbering carried over into the new certificate)

File: spec/requests/certificates/i_belong_dashboard_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    describe 'while unenrolled' do
+      before do
+        certificate
+        allow_any_instance_of(AuthenticationHelper).to receive(:current_user).and_return(user)
+      end
+
+      it 'redirects to I Belong public page' do
+        skip 'public page branch not merged yet, so i_belong_path not defined'
+
```
Comment (dcorking) [inline spec/requests/certificates/i_belong_dashboard_spec.rb:22]:
```suggestion
        pending 'public page branch not merged yet, so i_belong_path not defined'

```
I think I meant 'pending' here, because when the i-belong public page branch is merged, this spec will automatically fail and the pending line can be removed to make it pass (Hopefully someone will also notice the TODOs at the same time but 🤷 )

**Comment by dcorking** [comment]
I thought I already posted this question but it didn't save AFAICT:
should the 'in progress' badge in the new cert

Screenshot 2023-08-09 at 11 40 22

look more like this (in CS accelerator it floats to the right, in the same line as 'Remote course' and duration)?
Screenshot 2023-08-09 at 11 40 09

File: app/components/community_activity_component/community_activity_component.html.erb (hunk 1 above)
Comment (Caleb-T-Owens) [reply app/components/community_activity_component/community_activity_component.html.erb]: You're right, with my new condition, that became quite the chunky partial. There is no good way to have partials for a view component so I've refactored those if-else statements so its less indented and makes the control flow a little clearer to understand. I hope this is a good compromise

**Review by dcorking — DISMISSED** [review]
LGTM 🚂

**Review by dcorking — APPROVED** [review]
Still LGTM 🚂
